tests/test_conftest.py

```diff
@@ -15,17 +14,17 @@ def test_ndjson_data_generator(tmp_path):
         for filepath in [f for f in Path(target / key).iterdir()]:
             with open(filepath) as f:
                 first_new = json.loads(next(f))
-                for line in f:
-                    pass
-                last_new = json.loads(line)
+                *_, last_new = f
+                last_new = json.loads(last_new)
             with open(f"{Path(MOCK_DATA_DIR)}/{key}/{filepath.name}") as f:
                 first_line = next(f)
                 first_ref = json.loads(first_line)
                 # handling patient file of length 1:
-                line = first_line
-                for line in f:
-                    pass
-                last_ref = json.loads(line)
+                try:
+                    *_, last_ref = f
+                    last_ref = json.loads(last_ref)
+                except Exception:
+                    last_ref = first_ref
```

Comment:
What's with this exception here? That's new behavior yeah? Is there a more specific exception class to reference?

Reply:
if you try to unpack an iterator and its empty, you get an error - the docs didn't specify which. I can do some research.

Reply:
Speaking of, it's interesting to me that `ruff` doesn't complain about a loose `Exception` catch. Which... I don't mind. There are plenty of valid reasons for it.

Can you check if it _would_ complain about a bare `except`? Like if you did `except:` just alone -- I would want it to treat that as a flaggable error (and probably not an auto-fix one).

Reply:
it did, that's how we ended up here in the first place.

Reply:
There wasn't a bare-except here before. But maybe during PR work you had one?

Reply:
yeah, it happened midstream